### PR TITLE
Add and pass tests

### PR DESCRIPTION
**Pull request by pietgeursen**
@ahdinosaur Review thiiiiiissssss.

### CODE REVIEW

File: test/dispatch.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+test('Calling dispatch triggers update function with action passed to dispatch.', function (t) {
+  var initialModel = {initial: true}
+  var expectedAction = {type: 'DISPATCHED'}
+  var app = {
+    init: function () {
+      return {model: initialModel}
+    },
+    update: function (model, action) {
+      t.deepEqual(action, expectedAction, 'action passed to update is the action passed to dispatch')
+      t.end()
```
Comment (ahdinosaur) [inline test/dispatch.js]: can we also listen to the `actions` stream returned from `start(app)` for the action?

File: test/dispatch.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      return {model: model}
+    },
+    view: function (model, dispatch) {
+      dispatch(expectedAction)
```
Comment (ahdinosaur) [inline test/dispatch.js]: i wonder if we should also test for dispatching sometime later, even just after `process.nextTick`.

File: test/dispatch.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+test('Calling dispatch triggers update function with action passed to dispatch.', function (t) {
+  var initialModel = {initial: true}
+  var expectedAction = {type: 'DISPATCHED'}
+  var app = {
+    init: function () {
+      return {model: initialModel}
+    },
+    update: function (model, action) {
+      t.deepEqual(action, expectedAction, 'action passed to update is the action passed to dispatch')
```
Comment (ahdinosaur) [inline test/dispatch.js]: for these checks you can use `t.equal`, they should be `===` since they should be the same object.

File: test/run.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    view: function (model, dispatch) {
+      return inu.html`<div></div>`
+    },
+    run: function (effect) {
+      return pull.values([expectedAction])
+    }
+  }
+  inu.start(app)
+})
```
Comment (ahdinosaur) [inline test/run.js]: also maybe test for listening to actions stream in `run`?

File: test/update.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+var test = require('tape')
+
+var inu = require('../')
+var pull = inu.pull
+
+test('models returned by update are emitted by the model stream', function (t) {
+  var initialModel = {initial: true}
+  var expectedModel = {type: 'EXPECTED_MODEL'}
```
Comment (ahdinosaur) [inline test/update.js]: this looks like an action or effect :wink: (not that it really matters...)

File: test/update.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    view: function (model, dispatch) {
+      dispatch()
+      return inu.html`<div></div>`
+    }
+  }
+  var streams = inu.start(app)
+
+  pull(streams.models(), pull.take(2), pull.collect(function (err, models) {
+    t.false(err)
```
Comment (ahdinosaur) [inline test/update.js]: `t.error` ?

**Comment by ahdinosaur** [comment]
@pietgeursen hey, thanks for the sweet tests. :tada: made some comments.

some thoughts:

- maybe rename `initialRender` test to `view`, to match your naming here.
- i like how you used the returned models stream in your model test, maybe we should be checking these streams are correct in more of these tests?
- missing tests to (i think)  get 100% code coverage (not that 100% means we've deeply tested everything, just touched a bit on everything):
  - add test to listen to actions stream within `run`
  - add test to call `stop` returned by `start` and expect all streams to disconnect and nothing to happen anymore.

**Comment by pietgeursen** [comment]
@ahdinosaur Ok, I've added more tests and added your suggestions. There's still more to go with checking all the streams are emitting but this seems like a good start.

There are two skipped tests in the init test that I think should pass. Should they?

**Comment by ahdinosaur** [comment]
@pietgeursen churr purrr :smile_cat:

**Pull request merged**